gui/blueprint.lua

```diff
@@ -84,6 +85,67 @@ function ActionPanel:get_area_text()
     return ('%dx%dx%d (%d tile%s)'):format(width, height, depth, tiles, plural)
 end
 
+NamePanel = defclass(NamePanel, ResizingPanel)
+NamePanel.ATTRS{
+    name='blueprint',
+}
+function NamePanel:init()
+    self:addviews{
+        widgets.EditField{
+            view_id='name',
+            frame={t=0,h=1},
+            key='CUSTOM_N',
+            active=false,
+            text=self.name,
+            on_change=self:callback('detect_name_collision'),
+        },
+        widgets.Label{
+            view_id='name_help',
+            frame={t=1,l=2},
+            text={{text=self:callback('get_name_help', 1),
+                   pen=self:callback('get_help_pen')}, '\n',
+                  {text=self:callback('get_name_help', 2),
+                   pen=self:callback('get_help_pen')}}
+        },
+    }
+
+    self:detect_name_collision()
+end
+function NamePanel:detect_name_collision()
+    -- don't let base names start with a slash - it would get ignored by
+    -- the blueprint plugin later anyway
```

Comment:
Names that _end_ with a slash don't seem to be detected - I was able to save to `foo/` twice without getting a warning. I haven't looked into the logic much yet, but maybe `normalizePath()` in the loop below would help?

(This is very useful, though! I wouldn't have checked this case if not for https://github.com/DFHack/dfhack/pull/1891)

Reply:
fixed. Good catch! For the record, we do call `normalizePath()` before comparing filenames. it even prevents you from entering a non-normalized path (try typing two slashes in a row)